src/app/modules/common/services.js

```diff
@@ -546,18 +579,62 @@ angular.module('BloomLibraryApp.services', ['restangular'])
-        this.getTags = function () {
-            return [            
-                // Replicated from Bloom.Book.RuntimeInformationInjector.AddUISettingsToDom().
-                // Eventually this will be user-extensible and retrieved using a query on some new table.
-                "Agriculture", "Animal Stories", "Business", "Culture", "Community Living", "Dictionary", "Environment", "Fiction", "Health", "How To", "Math", "Non Fiction", "Spiritual", "Personal Development", "Primer", "Science", "Story Book", "Traditional Story"
-            ].sort();
+    .service('parseAngularService', ['$rootScope', '$q', '$filter', 'errorHandlerService', function($rootScope, $q, $filter, errorHandlerService) {
+        this.parsePromiseToAngular = function(parsePromise, functionRef) {
+            // query.find and other queries return a parse.com promise, but it is not quite the same api as
+            // as an angularjs promise. Instead, translate its find and error functions using the
+            // angularjs promise.
+
+            var defer = $q.defer(); // used to implement angularjs-style promise
+
+            parsePromise.then(function(results) {
+                var objects = new Array(results.length);
+                for (i = 0; i < results.length; i++) {
+                    objects[i] = results[i].toJSON();
+                }
+                // languageList = objects;
+                // See the discussion in getFilteredBookRange of why the $apply is used. I haven't tried
+                // NOT using it in this context.
+                $rootScope.$apply(function () { defer.resolve(objects); });
+            },
+            function (error) {
+                errorHandlerService.handleParseError(functionRef || 'unspecified', error);
+                defer.reject(error);
+            });
+
+            return defer.promise;
         };
-        this.getDisplayName = function(tagId) {
-            return tagId;
+    }])
+    .service('tagService', ['parseAngularService', function(parseAngularService) {
+        this.getTags = function (category) {
+            var tagQuery = new Parse.Query('tag');
+            tagQuery.descending("usageCount");
+            tagQuery.limit(1000);
+
+            //If a category is specified, only get tags of that category
+            if(category) {
+                tagQuery.startsWith('name', category + '.');
+            }
+
+            var parsePromise =  tagQuery.find();
+            return parseAngularService.parsePromiseToAngular(parsePromise, 'getTags');
         };
-    })
+        //Remove category (i.e. 'category.') and put spaces before camel-case type
+        this.getDisplayName = function(tag) {
+            try {
+                var prefixRegex = /[a-z]+\./;
+                var camelRegex = /([a-z])([A-Z])/g;
+
+                var ret = tag.replace(prefixRegex, "").replace(camelRegex, function (match, p1, p2) {
+                    return p1 + ' ' + p2;
```

Comment:
I don't know. John Hatton did not specify on that. I guess I assumed that such a tag would have been intentional on the librarian's part.